flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/expressions/time.scala

```diff
@@ -300,3 +300,25 @@ case class ToTimestampLtz(
 
   override private[flink] def resultType = SqlTimeTypeInfo.TIMESTAMP
 }
+
+case class DateAdd(
```

Comment:
This stack is deprecated, please remove this code. You need to add the type inference and return type inference to `BuiltInFunctionDefinitions`,